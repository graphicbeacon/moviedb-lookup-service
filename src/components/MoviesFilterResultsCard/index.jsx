import React from 'react';

const MoviesFilterResultsCard = ({
  original_title: originalTitle,
  overview
}) => (
  <div data-testid="resultsCard" className="card">
    <div className="card-body">
      <h3 data-testid="resultsCardTitle" className="card-title">
        {originalTitle}
      </h3>
      <p data-testid="resultsCardDesc" className="card-text">
        {overview}
      </p>
    </div>
  </div>
);

export default MoviesFilterResultsCard;
