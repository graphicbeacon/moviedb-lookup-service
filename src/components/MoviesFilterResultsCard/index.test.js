import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, getByTestId } from 'react-testing-library';

import MoviesFilterResultsCard from './index';

describe('<MoviesFilterResultsCard />', () => {
  test('should match generated snapshot', () => {
    const tree = TestRenderer.create(<MoviesFilterResultsCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render movie title and description', () => {
    const { container } = render(
      <MoviesFilterResultsCard
        original_title="movie original title"
        overview="movie overview details"
      />
    );
    const title = getByTestId(container, 'resultsCardTitle');
    const description = getByTestId(container, 'resultsCardDesc');

    expect(title.textContent).toBe('movie original title');
    expect(description.textContent).toBe('movie overview details');
  });
});
