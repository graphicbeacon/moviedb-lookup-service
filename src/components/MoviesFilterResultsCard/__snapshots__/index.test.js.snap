// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesFilterResultsCard /> should match generated snapshot 1`] = `
<div
  className="card"
  data-testid="resultsCard"
>
  <div
    className="card-body"
  >
    <h3
      className="card-title"
      data-testid="resultsCardTitle"
    />
    <p
      className="card-text"
      data-testid="resultsCardDesc"
    />
  </div>
</div>
`;
